[FIX] base: fix autoresize and exclude gif

nle-odoo opened the pull request:
Odoo may resize attachment image with side larger than 1920 pixels.

But for animated gifs, this resizement seems to in general increase size
file which is not what we want (in some case making it grow from 3MB to
60 MB).

So with this change, we only resize and optimize images that are not
gifs.

Reasoning: pillow doesn't seem to resize GIF (and seems to only increase
their size, especially animated GIF, because each frame is not
optimized) so we should just not touch them.

Note:

- currently tiff were not resized (because of a mimetype typo)
- currently image dimensions were not resized (from our test, resizing
  on the dimension does not change the size much, quality is most
  important).

both of these issue have been solved in this commit.

opw-[2897291](https://www.odoo.com/web#id=2897291&view_type=form&model=project.task)

Note: In the ticket case, a 4MB gif would grow to 60 MB. Here is an example of a 300 KB gif that is getting saved as 3 MB:

![bignyan](https://user-images.githubusercontent.com/9977887/179728314-8255e120-dedd-4457-93a7-c4d025ca984f.gif)

robodoo commented:
[Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/96307).

xmo-odoo commented on odoo/addons/base/models/ir_attachment.py:
```diff
@@ -316,7 +316,10 @@ def _postprocess_contents(self, values):
                     if w > nw or h > nh:
                         img.resize(nw, nh)
```
By the by, it looks like the resizing isn't even applied (`Image.resize` returns a new image).

Which I guess is sorta good news for animated gifs, because `Image.resize` doesn't understand multi-frame images, so it removes them all. For multi-frame images (animated gifs) we'd need to extract the individual frames, resize them, then recompose the gif.

nle-odoo commented:
> Maybe we should disable this entire thing for gifs?

I think it might be better, I've changed the PR so we don't do it for animated GIF. In any case I think animated gif higher than 1920 must not happen too often.

xmo-odoo reviewed (approved):
Seems good to me, I'd just like the commit message updated with the additional decisions and the reasoning.

nle-odoo commented:
Thanks, I did some more test and I removed the resize of all GIF completely. It always increased the size from my test even on non-animated ones, eg. 9 to 14 MB on pillow 5, improved to 9 to 9.1 MB on pillow 9.2 version.

I fixed the resize to not leave dead code.

xmo-odoo commented:
:+1: anything more to change or is this GTG?

nle-odoo commented:
good to go, thanks

robodoo r+
